File: backend/crawl/usline.js
const axios = require('axios');
const cheerio = require('cheerio');

async function crawlUsline() {
    try {
        const url = 'https://www.usline.kr/news/articleList.html?view_type=sm';

        // 요청 헤더 설정
        const headers = {
            'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36',
            'Referer': 'https://www.google.com/', // 검색을 통해 접근한 것처럼 보이게 설정
        };

        // 웹페이지 요청
        const { data } = await axios.get(url, { headers });

        // HTML 파싱
        const $ = cheerio.load(data);
        const newsList = $('section#section-list div.view-cont');

        const result = [];

        newsList.each((_, el) => {
            const titleTag = $(el).find('h4 a');
            // 제목 및 링크크
            let title = titleTag.text().trim();
            let link = 'https://www.usline.kr' + titleTag.attr('href');

            // 날짜
            let date = null;
            $(el).find('em').each((_, em) => {
                const match = $(em).text().trim().match(/(\d{4}\.\d{2}\.\d{2}) \d{2}:\d{2}/);
                if (match) {
                    date = match[1];
                    return false; // 첫 번째 일치하는 날짜만 사용
                }
            });

            result.push({ title, link, date });
        });

        console.log({ "유스라인(Usline)": result });
        return result;
    } catch (error) {
        console.error('크롤링 실패:', error.message);
        return [];
    }
}

// 약 0초
// crawlUsline();
module.exports = crawlUsline;

File: backend/crawl/usline.py
import requests
import re
from bs4 import BeautifulSoup

def get_data():
    url = 'https://www.usline.kr/news/articleList.html?view_type=sm'

    response = requests.get(url)

    if response.status_code == 200:
        soup = BeautifulSoup(response.text, 'html.parser')

        news_list = soup.select('section#section-list div.view-cont')

        result = []
        for news in news_list:
            title_tag = news.select_one('h4 a')
            title = title_tag.text.strip()

            # <em>사회</em><em>박병수 편집국장</em><em>2024.10.30 04:16</em>
            date = None
            for em in news.find_all('em'):
                match = re.match(r'(\d{4}\.\d{2}\.\d{2}) \d{2}:\d{2}', em.text)
                if match:
                    date = match.group(1)
                    break

            link = "https://www.usline.kr" + title_tag.get('href')

            #print(f"제목: {title}")
            #print(f"날짜: {date}")
            #print(f"링크: {link}")
            #print("-" * 100)
            dict_data = {"title": title, "link": link, "date": date}
            result.append(dict_data)

        return {"유스라인(Usline)": result}
    else:
        print(f"Failed to fetch the page, status code: {response.status_code}")
        return {"Error": response.status_code}